Clean compath usage and add prep_sfc job to rocoto mesh

KateFriedman-NOAA opened the pull request:
# Description

This PR covers several issues related to compath usage, COM variables, and adding the `prep_sfc` (formerly `emcsfc_sfc_prep`) job into the rocoto mesh for testing it. The following changes are in this PR:

- add the `prep_sfc` job into the rocoto mesh
  - copy the `prep_sfc` scripts from the UFS_UTILS repository and update them to match recent optimizations and standardization work in global-workflow
    - `dev/jobs/prepsfc.sh` (new)
    - `jobs/JGLOBAL_ATMOS_PREP_SFC` (renamed `jobs/JGLOBAL_ATMOS_EMCSFC_SFC_PREP`)
    - `scripts/exglobal_prep_sfc.sh` (renamed `scripts/exemcsfc_global_sfc_prep.sh` from UFS_UTILS)
    - `ush/prep_sfc_ice_blend.sh` (renamed `ush/emcsfc_ice_blend.sh` from UFS_UTILS)
    - `ush/prep_sfc_snow.sh` (renamed `ush/emcsfc_snow.sh` from UFS_UTILS)
  - add `prep_sfc` job into workflow configuration
  - add `prep_sfc` scripts in `CODEOWNERS` file with @GeorgeGayno-NOAA as owner (also remove from `.gitignore` file)
  - create `parm/config/gfs/config.prep_sfc`
  - add `prep_sfc` case block to `config.resources`
  - add `DO_PREP_SFC` switch in `config.base`
- COM and `compath.py` updates:
  - update COM variables in prep job `dev/jobs/prep.sh`
    - rename `COMIN_OBSDMP` to `COMIN_OBSPROC` to match other conventions
    - add `COMIN_TCVITAL` declaration
    - update `cp` to `cpfs`
    - add other COM variable declarations that are needed
- delete old ecFlow prod definition files (`ecf/defs/gfs_CC.def`)  and bring one from ops in `ecf/defs/gfs_v16_3.def`
- update some `prep_step` instances to `source` instead of `.`

Resolves #289 
Refs #2451
Resolves #3638

# Type of change

Bugzilla resolutions and feature request

# Change characteristics

- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO

# How has this been tested?

- CI suite on WCOSS2 with `DO_PREP_SFC="NO"`
- `C96_atm3DVar_extended` test case with `DO_PREP_SFC="YES"` (did not run `prep_sfc` job completely due to lack of new global AFWA data in GDA for test case time period, need to retest with cycled ICs after 20250326; job ran fine until point of missing input)

KateFriedman-NOAA committed:
Add emcsfc_sfc_prep job to rocoto mesh

Refs #3638

KateFriedman-NOAA committed:
COM updates to address compath.py usage

- rename COM_OBSDMP to COM_OBSPROC
- rename COMINsyn to COM_SYNDAT
- rename COMINecmwf to COM_ECMWF
- rename COMINukmet to COM_UKMET
- rename COMINnam to COM_NAM
- move all compath.py usage into config.com and
define all COM variables that look outside package
within a NCO block in config.com
- some script cleanup

Refs #289

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Make some shellcheck fixes to emcsfc_sfc_prep scripts

Refs #3638

KateFriedman-NOAA committed:
Shellcheck updates to ush/getdump.sh

KateFriedman-NOAA committed:
Updated to COM variables in prep.sh

Refs #2451
Refs #289

KateFriedman-NOAA committed:
An assortment of small updates

- some COM variable updates/fixes
- some shellcheck updates
- update COM for COMINtcvital and add to config.com

Refs #2451
Refs #289

KateFriedman-NOAA committed:
Rename emcsfc_sfc_prep job to prep_sfc job

Remove old ecf definition files and pull in def file from ops

Refs #3638

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Change sfcprep to prepsfc

Refs #3638

KateFriedman-NOAA committed:
Source prep_step and fix COM variable

Refs #3638

KateFriedman-NOAA committed:
Add prep_sfc scripts to CODEOWNERS file

Refs #3638

KateFriedman-NOAA committed:
Updates to get prep_sfc job running

- fix bug in gfs_tasks.py
- export gcycle in jobs/JGLOBAL_ATMOS_PREP_SFC
- replace JCAP and JCAP_ENKF with CRES and
CRES_ENS respectively

Refs #3638

KateFriedman-NOAA committed:
Address shellcheck warnings

Refs #289
Refs #3638

KateFriedman-NOAA committed:
Address shellcheck warning in prep_sfc_ice_blend.sh

Refs #3638

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

aerorahul commented on dev/jobs/prepsfc.sh:
```diff
@@ -0,0 +1,21 @@
+
+###############################################################
+# Source FV3GFS workflow modules
+. "${HOMEgfs}/ush/load_fv3gfs_modules.sh"
```
```suggestion
source "${HOMEgfs}/ush/load_fv3gfs_modules.sh"
```

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -131,21 +146,23 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
     if [[ ${err} -ne 0 ]]; then
-       echo "FATAL ERROR: Global prep job failed!"
-       exit 1
+       err_chk "FATAL ERROR: Global prep job failed!"
```
```suggestion
       err_exit "JOBSPROC_GLOBAL_PREP job failed!"
```
per @DavidHuber-NOAA 's new issue he has opened.

aerorahul replied:
Though, it also requires `export err` on line 147

KateFriedman-NOAA replied:
Updated @ 7aa4f5982

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -131,21 +146,23 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
     if [[ ${MAKE_NSSTBUFR:-"NO"} = "NO" ]]; then
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        if [[ ${DONST} = "YES" ]]; then
+           ${NCP} "${COMIN_OBSPROC}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```
```suggestion
           cpreq "${COMIN_OBSPROC}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -131,21 +146,23 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
+        ${NCP} "${COMIN_OBSPROC}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
+        if [[ ${DONST} = "YES" ]]; then
+           ${NCP} "${COMIN_OBSPROC}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
+        fi
```
```suggestion
        cpreq "${COMIN_OBSPROC}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
        cpreq "${COMIN_OBSPROC}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
        if [[ ${DONST} = "YES" ]]; then
           cpreq "${COMIN_OBSPROC}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
        fi
```

aerorahul commented on jobs/JGDAS_ATMOS_GEMPAK_META_NCDC:
```diff
@@ -47,8 +44,8 @@ if (( cyc%12 == 0 )); then
-export COMINukmet="${COMINukmet:-$(compath.py "${envir}/ukmet/${ukmet_ver}")/ukmet}"
-export COMINecmwf="${COMINecmwf:-$(compath.py "${envir}/ecmwf/${ecmwf_ver}")/ecmwf}"
+declare_from_tmpl -rx COMIN_ECMWF:COM_ECMWF_TMPL
+declare_from_tmpl -rx COMIN_UKMET:COM_UKMET_TMPL
```
I am not sure pushing this to `config.com` is going to go over well during EE2.  Keeping this here allowing to change the source of outside data provides greater flexibility to NCO

KateFriedman-NOAA replied:
Good point, have reverted this @ 7aa4f5982

aerorahul commented on jobs/JGFS_ATMOS_GEMPAK_META:
```diff
@@ -44,12 +44,9 @@ export DBN_ALERT_TYPE=GFS_METAFILE
-export DBNROOT=${DBNROOT:-${UTILROOT}/fakedbn}
+declare_from_tmpl -rx COMIN_UKMET:COM_UKMET_TMPL
+declare_from_tmpl -rx COMIN_ECMWF:COM_ECMWF_TMPL
+declare_from_tmpl -rx COMIN_NAM:COM_NAM_TMPL
```
I would keep the `compath.py` statements here as is, following the same argument above

KateFriedman-NOAA replied:
Reverted @ 7aa4f5982

aerorahul commented on parm/config/gfs/config.prepsfc:
```diff
@@ -0,0 +1,11 @@
+echo "BEGIN: config.prepsfc"
+
+# Get task specific resources
+. "${EXPDIR}/config.resources" prepsfc
```
```suggestion
source "${EXPDIR}/config.resources" prepsfc
```

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,198 @@
+# code can't run without an ice analysis.
+#-----------------------------------------------------------------------
+
+if ((err != 0)); then
```
```suggestion
if [[ ${err} -ne 0 ]]; then
```
(( )) statements don't get into the logs

KateFriedman-NOAA replied:
> (( )) statements don't get into the logs

Good to know, thank you!

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,198 @@
+
+export SNOW2MDLEXEC="${EXECgfs}/emcsfc_snow2mdl"
+
+CRES=$(echo "${CASE}" | cut -c2-)
```
Get rid of `CRES` and use `${CASE:1}` where `CRES` is used.

KateFriedman-NOAA replied:
Thanks, have made the update @ 7aa4f5982

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,198 @@
+# global cycling can't run without an snow analysis.
+#-----------------------------------------------------------------------
+
+if ((err != 0)); then
```
same as `if [[ ]]`

KateFriedman-NOAA replied:
Fixed @ 7aa4f5982, thanks!

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,198 @@
+    echo "COPY OLD ${CASE} SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
+  else
+    err_chk "FATAL ERROR: CURRENT AND 6-HR OLD ${CASE} SNOW MISSING"
```
```suggestion
    err_exit "CURRENT AND 6-HR OLD ${CASE} SNOW MISSING, ABORT!"
```

aerorahul commented on ush/prep_sfc_ice_blend.sh:
```diff
@@ -0,0 +1,132 @@
+#------------------------------------------------------------------------
+
+if [[ "${err}" -ne 0 ]]; then
+  echo "WARNING: ${pgm} completed abnormally. The old ice blend file will be used."
+  exit "${err}"
```
Leave a note here to allow handling of older file in the calling script and not to call `err_exit` here.

KateFriedman-NOAA replied:
Added note @ 5f66f9f89, thanks!

aerorahul commented on ush/prep_sfc_ice_blend.sh:
```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
```
``` suggestion
#!/usr/bin/env bash
```

aerorahul reviewed (commented):
some comments.

KateFriedman-NOAA committed:
Apply suggestions from code review

Refs #3638

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Address reviewer comments

- revert some COM updates
- replace NCP with cpfs
- replace CRES with CASE cut

Refs #289
Refs #3638

KateFriedman-NOAA committed:
Update OBSPROC and RTOFS COM variables

- change COMIN_OBSPROC back to COMINobsproc
- change COMIN_RTOFS to COMINrtofs

Refs #2451

KateFriedman-NOAA committed:
Change prepsfc instances to prep_sfc

Refs #3638

JessicaMeixner-NOAA reviewed (dismissed):
The cominrtofs changes are okay with me.  I only reviewed that one file though that was related to wave changes.

KateFriedman-NOAA committed:
Updates to prep_sfc scripts

- add comment to prep_sfc_ice_blend.sh about not using err_exit
- fix syntax for if-block in exglobal_prep_sfc.sh

Refs #3638

sbanihash reviewed (dismissed):
I have reviewed the changes in wave_prep /job and /scripts. These changes look ok. I approve these two changes regarding the wave related changes.

KateFriedman-NOAA committed:
Additional external COM updates

Refs #2451

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

aerorahul commented on .github/CODEOWNERS:
```diff
@@ -182,6 +183,8 @@ ush/parsing_namelists_MOM6.sh @WalterKolczynski-NOAA @aerorahul @junwang-noaa @j
 ush/parsing_namelists_WW3.sh @WalterKolczynski-NOAA @aerorahul @JessicaMeixner-NOAA @sbanihash
 ush/parsing_ufs_configure.sh @WalterKolczynski-NOAA @aerorahul @junwang-noaa
 ush/preamble.sh @WalterKolczynski-NOAA
+ush/prep_sfc_ice_blend.sh @GeorgeGayno-NOAA
+ush/prep_sfc_snow.sh @GeorgeGayno-NOAA
```
```suggestion
ush/prep_sfc_ice_blend.sh @GeorgeGayno-NOAA @KateFriedman-NOAA 
ush/prep_sfc_snow.sh @GeorgeGayno-NOAA @KateFriedman-NOAA 
```

aerorahul commented on .github/CODEOWNERS:
```diff
@@ -131,6 +131,7 @@ scripts/exglobal_marinebmat.py @guillaumevernieres @AndrewEichmann-NOAA
 scripts/exglobal_oceanice_products.py @GwenChen-NOAA @AminIlia-NOAA @ChristopherHill-NOAA
 scripts/exglobal_prep_emissions.py @bbakernoaa
 scripts/exglobal_prep_obs_aero.py @CoryMartin-NOAA
+scripts/exglobal_prep_sfc.sh @GeorgeGayno-NOAA
```
```suggestion
scripts/exglobal_prep_sfc.sh @GeorgeGayno-NOAA @KateFriedman-NOAA 
```

aerorahul commented on .github/CODEOWNERS:
```diff
@@ -59,7 +59,7 @@ jobs/JGLOBAL_*ARCH* @DavidHuber-NOAA
 jobs/JGLOBAL_ATMOS_ANALYSIS @RussTreadon-NOAA @CatherineThomas-NOAA
 jobs/JGLOBAL_ATMOS_ANALYSIS_CALC @RussTreadon-NOAA @CatherineThomas-NOAA @CoryMartin-NOAA
-jobs/JGLOBAL_ATMOS_EMCSFC_SFC_PREP @GeorgeGayno-NOAA
+jobs/JGLOBAL_ATMOS_PREP_SFC @GeorgeGayno-NOAA
```
```suggestion
jobs/JGLOBAL_ATMOS_PREP_SFC @GeorgeGayno-NOAA @KateFriedman-NOAA 
```

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -78,16 +82,19 @@ fi
     if [[ ! -s ${ARCHSYND}/syndat_akavit ]]; then
         for file in syndat_akavit syndat_dateck syndat_stmcat.scr syndat_stmcat syndat_sthisto syndat_sthista ; do
-            cp "${COMINsyn}/${file}" "${ARCHSYND}"/.
+            cpreq "${COMIN_SYNDAT}/${file}" "${ARCHSYND}"/.
```
I think `COMINsyndat` is the right name, given that there is a package named `syndat`.
We should update every reference to `COMINsyndat` instead of `COMINsyn` (except in the TC package where it for some reason uses `COMINsyn`).  I definitely would not creating `COMIN_SYNDAT`

KateFriedman-NOAA replied:
I was going to leave it as `COMINsyndat` as well but this is a GFS folder that lives at the top of GFS com (`../com/gfs/v16.3/syndat`) alongside the `RUN.PDY` folders, so it's internal to the GFS. I've been waffling still on changing it back...thoughts?

KateFriedman-NOAA replied:
Discussed offline and agreed to change back to `COMINsyndat`. Will do that now.

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -96,30 +103,38 @@ if [[ ${PROCESS_TROPCY} = "YES" ]]; then
 # Generate prepbufr files from dumps or copy from OPS
-if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
-    if [[ ${ROTDIR_DUMP} = "YES" ]]; then
+if [[ ${MAKE_PREPBUFR:-"YES"} = "YES" ]]; then
+    if [[ ${ROTDIR_DUMP:-"YES"} = "YES" ]]; then
```
Why provide a default `YES` only here?

KateFriedman-NOAA replied:
"NO" doesn't work currently so we want it set to "YES".

KateFriedman-NOAA replied:
Will look at removing the `ROTDIR_DUMP` option in a separate PR.

KateFriedman-NOAA replied:
See issue #3693

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -96,30 +103,38 @@ if [[ ${PROCESS_TROPCY} = "YES" ]]; then
 # Generate prepbufr files from dumps or copy from OPS
-if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
-    if [[ ${ROTDIR_DUMP} = "YES" ]]; then
+if [[ ${MAKE_PREPBUFR:-"YES"} = "YES" ]]; then
```
Default should be `NO`, IMO.  Unless asked to create, let it copy the one from dump archive.

KateFriedman-NOAA replied:
No we want a prepbufr generated within the experiment because the prepbufr generation uses guess from the prior gdas cycle as input. If you pull in the prepbufr from the GDA (ops v16) then it will essentially wipe out the prior guess and introduce a prepbufr generated with guess from ops. We always want a prepbufr generated if we're cycling.
```
kate.friedman@clogin02:/lfs/h2/emc/ptmp/kate.friedman/comrot/RUNTESTS/COMROOT/C96_atm3DVar_extended> grep gdas logs/2021122100/gfs_prep.log | grep cp
+ 2.119s + cp /lfs/h2/emc/ptmp/kate.friedman/comrot/RUNTESTS/COMROOT/C96_atm3DVar_extended/gdas.20211220/18//model/atmos/history/gdas.t18z.atmf006.nc sgesprep
```

DavidHuber-NOAA replied:
`grep`-ing through obsproc and prepobs, I don't see references to `MAKE_PREPBUFR`.  I do see `MAKE_NSSTBUFR`, though.  I think `MAKE_PREPBUFR` is now assumed `YES`.  The `JOBSPROC_GLOBAL_PREP` job calls `exglobal_makeprepbufr.sh` without an `if` block.  We could consider removing it from the workflow.

KateFriedman-NOAA replied:
Ah, thanks for checking @DavidHuber-NOAA ! I'll remove it, thanks!

KateFriedman-NOAA replied:
Removed `MAKE_PREPBUFR` switch and use @ 538218478.

aerorahul commented on dev/workflow/rocoto/gfs_tasks.py:
```diff
@@ -66,17 +66,50 @@ def stage_ic(self):
+    def prep_sfc(self):
+
+        dependencies = None
+        if self.options['do_prep_sfc']:
```
This method should not be in the task list if this option is NO.
```suggestion
```
The remaining of the method will need to be left indented.

KateFriedman-NOAA replied:
Fixed, thanks!

aerorahul commented on dev/workflow/hosts/hera.yaml:
```diff
@@ -3,16 +3,16 @@ DMPDIR: '/scratch1/NCEPDEV/global/glopara/dump'
-COMINecmwf: /scratch1/NCEPDEV/global/glopara/data/external_gempak/ecmwf
-COMINnam: /scratch1/NCEPDEV/global/glopara/data/external_gempak/nam
-COMINukmet: /scratch1/NCEPDEV/global/glopara/data/external_gempak/ukmet
+COM_SYNDAT: '/scratch1/NCEPDEV/global/glopara/com/gfs/prod/syndat'
```
```suggestion
COMINsyndat: '/scratch1/NCEPDEV/global/glopara/com/gfs/prod/syndat'
```
Here and other host yamls

KateFriedman-NOAA replied:
Updated, thanks!

aerorahul commented on jobs/JGFS_ATMOS_CYCLONE_GENESIS:
```diff
@@ -38,7 +39,7 @@ export COMINgenvit=${COMOUT_ATMOS_GENESIS}
 export COMOUT=${COMOUT_ATMOS_GENESIS}
 
-export COMINsyn=${COMINsyn:-$(compath.py "${envir}/com/gfs/${gfs_ver}")/syndat}
+export COMINsyn=${COMIN_SYNDAT}
```
```suggestion
export COMINsyn=${COMINsyndat:-$(compath.py "${envir}/com/gfs/${gfs_ver}")/syndat}
```

aerorahul commented on ush/prep_sfc_snow.sh:
```diff
@@ -0,0 +1,198 @@
+source prep_step
+
+rm -f ./fort.41
+cat > ./fort.41 << !
```
```suggestion
cat > ./fort.41 << EOF
```

aerorahul commented on ush/prep_sfc_snow.sh:
```diff
@@ -0,0 +1,198 @@
+  min_snow_depth=0.05
+  snow_cvr_threshold=50.0
+ /
+!
```
```suggestion
EOF
```

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/remove_compath_usage

* origin/develop:
  STY: Remove empty __init.py__ files. (#3691)
  Relocate config templates to `dev/` space in prep for EE2 (#3684)

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_CYCLONE_TRACKER:
```diff
@@ -44,8 +45,7 @@ export gfsdir=${COMIN_ATMOS_GRIB_0p25}
 export COMOUT=${COMOUT_ATMOS_TRACK}
-export COMINsyn=${COMINsyn:-$(compath.py ${envir}/com/gfs/${gfs_ver})/syndat}
-
+export COMINsyn=${COMIN_SYNDAT}
```
```suggestion
export COMINsyn=${COMINsyndat:-$(compath.py "${envir}/com/gfs/${gfs_ver}")/syndat}
```

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_CYCLONE_GENESIS:
```diff
@@ -27,6 +27,7 @@ export SCRIPTens_tracker=${SCRIPTens_tracker:-${HOMEens_tracker}/scripts}
 ##############################################
 YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COMOUT_ATMOS_GENESIS:COM_ATMOS_GENESIS_TMPL
 YMD=${PDY} HH=${cyc} GRID="0p25" declare_from_tmpl -rx COMIN_ATMOS_GRIB_0p25:COM_ATMOS_GRIB_GRID_TMPL
+declare_from_tmpl -rx COMIN_SYNDAT:COM_SYNDAT_TMPL
```
```suggestion
```

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_CYCLONE_TRACKER:
```diff
@@ -34,6 +34,7 @@ YMD=${PDY} HH=${cyc} declare_from_tmpl -rx \
   COMIN_ATMOS_GENESIS:COM_ATMOS_GENESIS_TMPL
 YMD=${PDY} HH=${cyc} GRID="0p25" declare_from_tmpl -rx \
   COMIN_ATMOS_GRIB_0p25:COM_ATMOS_GRIB_GRID_TMPL
+declare_from_tmpl -rx COMIN_SYNDAT:COM_SYNDAT_TMPL
```
```suggestion
```

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_FSU_GENESIS:
```diff
@@ -38,16 +38,15 @@ export PYTHONPATH=${USHens_tracker}/FSUgenesisPY:${PYTHONPATH}
 ##############################################
 YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COMOUT_ATMOS_GENESIS:COM_ATMOS_GENESIS_TMPL
 YMD=${PDY} HH=${cyc} GRID="0p25" declare_from_tmpl -rx COMIN_ATMOS_GRIB_0p25:COM_ATMOS_GRIB_GRID_TMPL
+declare_from_tmpl -rx COMIN_SYNDAT:COM_SYNDAT_TMPL
```
```suggestion
```

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_FSU_GENESIS:
```diff
@@ -38,16 +38,15 @@ export PYTHONPATH=${USHens_tracker}/FSUgenesisPY:${PYTHONPATH}
 export COMOUT=${COMOUT_ATMOS_GENESIS}
-
-export COMINsyn=${COMINsyn:-$(compath.py "${envir}/com/gfs/${gfs_ver}")/syndat}
+export COMINsyn=${COMIN_SYNDAT}
```
```suggestion
export COMINsyn=${COMINsyndat:-$(compath.py "${envir}/com/gfs/${gfs_ver}")/syndat}
```

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Remove COM_SYNDAT_TMPL and put COMINsyndat back

Refs #2451

KateFriedman-NOAA committed:
Change COMINsyndat to just COMINsyn

Refs #2451

KateFriedman-NOAA committed:
Update to prep_sfc in gfs_tasks.py

Refs #3638

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Fix pynorms warning

Refs #3638

aerorahul commented on dev/workflow/rocoto/gfs_tasks.py:
```diff
@@ -66,17 +66,49 @@ def stage_ic(self):
 
+    def prep_sfc(self):
+
+        dependencies = None
```
```suggestion
```
Is irrelevant, overwritten by line 79

aerorahul commented on dev/workflow/rocoto/gfs_tasks.py:
```diff
@@ -66,17 +66,49 @@ def stage_ic(self):
+                     'resources': resources,
+                     'dependency': dependencies,
+                     'envars': self.envars,
+                     'cycledef': cycledef,
```
```suggestion
                     'cycledef': self.run,
```

aerorahul commented on dev/workflow/rocoto/gfs_tasks.py:
```diff
@@ -66,17 +66,49 @@ def stage_ic(self):
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
+
+        cycledef = self.run
+
```
```suggestion
```

aerorahul commented on jobs/JGLOBAL_ATMOS_PREP_SFC:
```diff
@@ -0,0 +1,55 @@
+GDATE=$(date --utc +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
+gPDY=${GDATE:0:8}
+gcyc=${GDATE:8:2}
+export gcycle="t${gcyc}z"
```
I understand this is a convenient short-hand, but `${gcyle}` is no less characters than `t${gyc}z`.  IMO, it just is a re-definition and we should avoid it.

KateFriedman-NOAA replied:
Updated @ f40491231. I also updated some `${cycle}="t${cyc}z` references.

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+  fi  # check of missing 6-hr snow file
+fi  # return code check
+
+#-----------------------------------------------------------------------
+# The GFS step does not run the enkf stuff.  So there is no need to
+# create an enkf snow file.
+#-----------------------------------------------------------------------
+
+if test "${RUN}" = "gfs"
+then
+  exit 0
+fi
```
This may depend on early cycle EnKF `RUN=enkfgfs` is enabled or not.

KateFriedman-NOAA replied:
Updated to check if the ENKF is running for the specific RUN. Committed @ 9d6acea00.

aerorahul commented on ush/getdump.sh:
```diff
@@ -4,35 +4,34 @@ COMPONENT=${COMPONENT:-atmos}
-cyc=$(echo $CDATE |cut -c 9-10)
-prefix="$RUN.t${cyc}z."
-
+cyc=$(echo "${CDATE}" |cut -c 9-10)
```
```suggestion
```
`cyc` is a workflow variable set higher up.  Infact, it is used on line 7,8.
I don't think it is needed here.

aerorahul replied:
And with that, `CDATE` will no longer be required as argument 1 to this script.  Consider removing it.

KateFriedman-NOAA replied:
I have removed that `cyc` line (I agree, it shouldn't be there) but currently need the ability to send in `gPDY` and `gcyc` so while I have removed `CDATE` from the script I have also updated it to ingest a new input for `YMD` (`PDY/gPDY`) and `HH` (`cyc/gcyc`). The `getdump.sh` script needs a rewrite (see #183) so this will be fine for now. Committed @ b6e9c3278.

DavidHuber-NOAA commented on dev/jobs/prep.sh:
```diff
@@ -129,23 +142,25 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr"               "${COMOUT_OBS}/${OPREFIX}prepbufr"
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        cpreq "${COMINobsproc}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
+        cpreq "${COMINobsproc}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
+        if [[ ${DONST} = "YES" ]]; then
+           cpreq "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```
`cpfs` when copying to `COM`

```suggestion
        cpfs "${COMINobsproc}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
        cpfs "${COMINobsproc}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
        if [[ ${DONST} = "YES" ]]; then
           cpfs "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```

KateFriedman-NOAA replied:
Lol pretty sure I had these as `cpfs` and then @aerorahul requested `cpreq`. But I agree, should be `cpfs`.

aerorahul replied:
Ok.  Let me explain why `cpreq` is the right one here.
Even though this is a `COMOUT` space, these files are copied from `COMINobsproc` into our `COMOUT` space and are required for the GSI to run in the analysis step.
We could keep this `cpfs` which if it doesn't find these files will allow `prep.sh` to continue and pass.  However, when the GSI comes along in `anal.sh`, it will fail.  So, by using `cpreq` here, we are catching the error much earlier.

DavidHuber-NOAA replied:
`cpfs` also has a check that the copy was completed successfully, so it is an implied `cpreq` + `cp; mv` operation:

```
cp $1 $cpdstfile.cptmp

if [ $? -ne 0 ] ; then
   err_exit "$1 is missing or was not copied successfully."
fi

${FSYNC} $cpdstfile.cptmp
if [ $? -ne 0 ]; then
   >&2 echo "WARNING: ${FSYNC} $cpdstfile.cptmp failed."
fi

mv $cpdstfile.cptmp $cpdstfile
if [ $? -ne 0 ] ; then
   err_exit "$cpdstfile.cptmp is missing or was not moved successfully."
fi
```

KateFriedman-NOAA replied:
> However, when the GSI comes along in anal.sh, it will fail. So, by using cpreq here, we are catching the error much earlier.

I like that however, since we don't run the prep job in the GFS suite in ops (upstream obsproc does) we should have this handled within the analysis job if needed.

KateFriedman-NOAA replied:
Discussed offline, will use `cpfs`.

DavidHuber-NOAA commented on dev/jobs/prep.sh:
```diff
@@ -129,23 +142,25 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
     if [[ ${MAKE_NSSTBUFR:-"NO"} = "NO" ]]; then
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        if [[ ${DONST} = "YES" ]]; then
+           cpreq "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```
```suggestion
           cpfs "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

DavidHuber-NOAA commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${BLENDED_ICE_FILE_PREV}" ]]; then
+    echo "Copy old ice blend file to current directory"
+    cpfs "${BLENDED_ICE_FILE_PREV}" "${COMOUT_OBS}/${BLENDED_ICE_FILE}"
```
Should probably set `err` back to 0.

```suggestion
    cpfs "${BLENDED_ICE_FILE_PREV}" "${COMOUT_OBS}/${BLENDED_ICE_FILE}"
    export err=0
```

DavidHuber-NOAA commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ${CASE} SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
```
```suggestion
    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
    export err=0
```

DavidHuber-NOAA commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+    echo "COPY OLD ENKF SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
+  else
+    err_chk "FATAL ERROR: CURRENT AND 6-HR OLD ENKF SNOW MISSING"
```
No need to add `FATAL ERROR`.  `err_chk` and `err_exit` prepend that.  Also, prefer `err_exit` over `err_chk`

```suggestion
    err_exit "CURRENT AND 6-HR OLD ENKF SNOW MISSING"
```

KateFriedman-NOAA replied:
Committed @ e88b3022b.

aerorahul commented on dev/jobs/prep.sh:
```diff
@@ -129,23 +142,25 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr"               "${COMOUT_OBS}/${OPREFIX}prepbufr"
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        cpfs "${COMINobsproc}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
+        cpfs "${COMINobsproc}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
+        if [[ ${DONST} = "YES" ]]; then
+           cpfs "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
+        fi
```
I explained why I recommended `cpreq` here.
These files are required in the `anal.sh` step of the GSI.  So, if they are not present here, then the GSI will fail in a later job.  Having a `cpreq` here allows us to trap that error much sooner.

DavidHuber-NOAA replied:
`cpfs` acts as `cpreq` as well, so the error will be trapped with `cpfs` just as well as `cpreq`.

DavidHuber-NOAA commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ENKF SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
```
```suggestion
    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
    export err=0
```

KateFriedman-NOAA replied:
Committed @ e88b3022b.

aerorahul commented on scripts/exglobal_prep_sfc.sh:
```diff
@@ -0,0 +1,196 @@
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ENKF SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
```
```suggestion
    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
    export err=0
```

KateFriedman-NOAA replied:
Committed @ e88b3022b.

aerorahul commented on ush/prep_sfc_snow.sh:
```diff
@@ -0,0 +1,198 @@
+ /
+EOF
+
+eval "${SNOW2MDLEXEC}" >> "${pgmout}" 2> errfile
```
Why is this `eval`?

KateFriedman-NOAA replied:
@GeorgeGayno-NOAA Why is this line using `eval`?

aerorahul replied:
I don't think it is necessary, and can be removed.

GeorgeGayno-NOAA replied:
> @GeorgeGayno-NOAA Why is this line using `eval`?

These scripts were first created many years ago. Maybe that was an NCO standard at the time? Feel free to remove it.

KateFriedman-NOAA replied:
Thanks @GeorgeGayno-NOAA for confirming! I have removed the `eval` @ b0b70d01b.

DavidHuber-NOAA commented on ush/getdump.sh:
```diff
@@ -4,35 +4,34 @@ COMPONENT=${COMPONENT:-atmos}
-cd $SOURCE_DIR
-if [ -s ${prefix}updated.status.tm00.bufr_d ]; then
+cd "${SOURCE_DIR}"
+if [ -s "${prefix}updated.status.tm00.bufr_d" ]; then
```
Prefer double braces:
```suggestion
if [[ -s "${prefix}updated.status.tm00.bufr_d" ]]; then
```

KateFriedman-NOAA replied:
Thought I'd gotten those, thanks! Committed @ e88b3022b.

KateFriedman-NOAA committed:
Replace cycle/gcycle with tcyc/gcycz

Refs #3638

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Update exglobal_prep_sfc.sh for enkf run

Refs #3638

KateFriedman-NOAA committed:
Updates to ush/getdump.sh

Refs #183

KateFriedman-NOAA committed:
Address reviewer comments

Refs #3638

KateFriedman-NOAA committed:
Remove MAKE_PREPBUFR switch and usage

Refs #3638

aerorahul reviewed (dismissed):
looks good to me.

KateFriedman-NOAA committed:
Remove eval from exec invocation in prep_sfc_snow.sh

Refs #3638

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

aerorahul committed:
use template, atparse and date in snow2mdl exec

KateFriedman-NOAA committed:
Merge pull request #2 from aerorahul/patch/3683

Use template, atparse and date in snow2mdl exec

KateFriedman-NOAA committed:
Add shellcheck disable SC2034 in prep_sfc_snow.sh

Refs #3638

KateFriedman-NOAA committed:
Remove disable and add exports in prep_sfc_snow.sh

Refs #3638

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA commented:
@aerorahul I'm having trouble finding your comment now but you noted somewhere in here that NCO would want the `&source_data` input files for `prep_sfc` copied into DATA before being used. How does this change look?
```
kate.friedman@clogin01:/lfs/h2/emc/global/noscrub/kate.friedman/git/feature-remove_compath_usage> gdis
diff --git a/parm/prep_sfc/snow2mdl.nml.tmpl b/parm/prep_sfc/snow2mdl.nml.tmpl
index 776f15278..96042e58e 100644
--- a/parm/prep_sfc/snow2mdl.nml.tmpl
+++ b/parm/prep_sfc/snow2mdl.nml.tmpl
@@ -1,8 +1,8 @@
 &source_data
   autosnow_file=""
-  nesdis_snow_file="@[IMS_FILE]"
+  nesdis_snow_file="./imssnow96.grb.grib2"
   nesdis_lsmask_file=""
-  afwa_snow_global_file="@[AFWA_GLOBAL_FILE]"
+  afwa_snow_global_file="./snow.usaf.grib2"
   afwa_snow_nh_file=""
   afwa_snow_sh_file=""
   afwa_lsmask_nh_file=""
diff --git a/ush/prep_sfc_snow.sh b/ush/prep_sfc_snow.sh
index 9cebb726d..a9610d28b 100755
--- a/ush/prep_sfc_snow.sh
+++ b/ush/prep_sfc_snow.sh
@@ -87,11 +87,13 @@ MODEL_SNOW_FILE=${MODEL_SNOW_FILE:-"snogrb_model"}
 OUTPUT_GRIB2=${OUTPUT_GRIB2:-".false."}  # grib 1 when false.
 
 #------------------------------------------------------------------------
-# Do a quick check of the ims data to ensure it exists and is
-# not corrupt.
+# Do a quick check of the ims data to ensure it exists and is not corrupt.
+# If available, copy to DATA.
 #------------------------------------------------------------------------
 
-if [[ ! -f ${IMS_FILE} ]]; then
+if [[ -f ${IMS_FILE} ]]; then
+  cpreq "${IMS_FILE}" "${DATA}/imssnow96.grb.grib2"
+else
   echo "WARNING: Missing IMS data. Will not run ${SNOW2MDLEXEC}."
   exit 7
 fi
@@ -101,31 +103,33 @@ fi
 # ims data has highest priority of all input data.
 #------------------------------------------------------------------------
 
-${WGRIB2} -d 1 "${IMS_FILE}"
+${WGRIB2} -d 1 "imssnow96.grb.grib2"
 err=$?
 if [[ ${err} -ne 0 ]]; then
   echo "WARNING: Corrupt IMS data. Will not run ${SNOW2MDLEXEC}."
   exit 9
 else
-  tempdate=$(${WGRIB2} -t "${IMS_FILE}" | head -1) || true
+  tempdate=$(${WGRIB2} -t "imssnow96.grb.grib2" | head -1) || true
   IMSDATE=${tempdate#*d=}
 fi
 
 #------------------------------------------------------------------------
 # Ensure AFWA data exists and is not too old.
+# If available, copy to DATA.
 #------------------------------------------------------------------------
 
 if [[ ! -f ${AFWA_GLOBAL_FILE} ]]; then
   echo "WARNING: Missing AFWS data. Will not run ${SNOW2MDLEXEC}."
   exit 3
 else
-  ${WGRIB2} -d 1 "${AFWA_GLOBAL_FILE}"
+  cpreq "${AFWA_GLOBAL_FILE}" "${DATA}/snow.usaf.grib2"
+  ${WGRIB2} -d 1 "snow.usaf.grib2"
   err=$?
   if [[ ${err} -ne 0 ]]; then
     echo "WARNING: Corrupt AFWS data. Will not run ${SNOW2MDLEXEC}."
     exit "${err}"
   else
-    tempdate=$(${WGRIB2} -d 1 -t "${AFWA_GLOBAL_FILE}")
+    tempdate=$(${WGRIB2} -d 1 -t "snow.usaf.grib2")
     AFWADATE=${tempdate#*d=}
     two_days_ago=$(date --utc -d "${IMSDATE:0:8} ${IMSDATE:8:2} - 48 hours" +%Y%m%d%H)
     if [[ ${AFWADATE} -lt ${two_days_ago} ]]; then
@@ -135,7 +139,9 @@ else
   fi
 fi
 
+#------------------------------------------------------------------------
 # Additional variables used in the namelist for &output_grib_time
+#------------------------------------------------------------------------
 export IMSYEAR=${IMSDATE:0:4}
 export IMSMONTH=${IMSDATE:4:2}
 export IMSDAY=${IMSDATE:6:2}
```

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/remove_compath_usage

* origin/develop:
  Invoke err_chk for fatal errors in the wave J-Jobs and ex-scripts

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

KateFriedman-NOAA committed:
Update static_data.yml

Fix wrong type set for sync list (should have been checkboxes
instead of a dropdown). Also reorder platforms based on
tier status.

KateFriedman-NOAA committed:
Update static_data.yml

Fix task list in static data issue template.

KateFriedman-NOAA committed:
Update static_data.yml

Fix syntax error in new checklist in static data issue template.

KateFriedman-NOAA committed:
Copy prep_sfc inputs to DATA before using

- Move source_data inputs (ims and afwa data files) to
DATA before job uses them.
- Also rename ims temp file to drop ".grb" from name.

Refs 3638

DavidHuber-NOAA commented on dev/jobs/prep.sh:
```diff
@@ -28,30 +28,32 @@ GDUMP="gdas"
 
-export MAKE_PREPBUFR=${MAKE_PREPBUFR:-"YES"}
-if [[ ! -d "${COMOUT_OBS}" ]]; then mkdir -p "${COMOUT_OBS}"; fi
+if [[ ! -d "${COMOUT_OBS}" ]]; then
+   mkdir -p "${COMOUT_OBS}"
+fi
```
`mkdir -p` does nothing if the directory already exists, so you don't need to wrap this (Note that this is not true if `-p` is not provided):

```suggestion
mkdir -p "${COMOUT_OBS}"
```

KateFriedman-NOAA replied:
Thanks @DavidHuber-NOAA !

DavidHuber-NOAA commented on dev/jobs/prep.sh:
```diff
@@ -78,16 +80,19 @@ fi
-    export COMINsyn=${COMINsyn:-$(compath.py gfs/prod/syndat)}
     export ARCHSYND=${ROTDIR}/syndat
-    if [[ ! -d ${ARCHSYND} ]]; then mkdir -p "${ARCHSYND}"; fi
+    if [[ ! -d ${ARCHSYND} ]]; then
+       mkdir -p "${ARCHSYND}"
+    fi
```
```suggestion
    mkdir -p "${ARCHSYND}"
```

DavidHuber-NOAA reviewed (dismissed):
Looks good.  Made a note on `mkdir -p`, but I'm OK with keeping the checks in place.

KateFriedman-NOAA committed:
Apply suggestions from code review

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

aerorahul reviewed (approved):
lgtm.
thanks for bringing this into the gw fold and cleaning up the compath usage.

KateFriedman-NOAA commented:
WCOSS2 CI completed successfully (minus one failed gempak job from known issue):
```
******** C48_ATM_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    May 20 2025 17:31:56    May 20 2025 18:56:20

******** C48mx500_3DVarAOWCDA_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    May 20 2025 17:31:58    May 20 2025 17:46:23
202103250000        Done    May 20 2025 17:31:58    May 20 2025 19:26:26

******** C48mx500_hybAOWCDA_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    May 20 2025 17:32:00    May 20 2025 17:46:29
202103250000        Done    May 20 2025 17:32:00    May 20 2025 19:01:42

******** C48_S2SW_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    May 20 2025 17:32:02    May 20 2025 18:56:32

******** C48_S2SWA_gefs_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    May 20 2025 17:32:07    May 20 2025 18:42:10

******** C96_atm3DVar_extended_3683 ********
202112210000    gfs_gempak_f147-f168                   196633420                DEAD                 -29         2        1854.0

******** C96C48_hybatmaerosnowDA_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    May 20 2025 17:32:12    May 20 2025 17:56:55
202112201800        Done    May 20 2025 17:32:12    May 20 2025 20:01:59
202112210000        Done    May 20 2025 17:32:12    May 20 2025 19:52:10

******** C96C48_hybatmDA_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    May 20 2025 17:32:14    May 20 2025 17:52:03
202112210000        Done    May 20 2025 17:32:14    May 20 2025 19:57:33
202112210600        Done    May 20 2025 17:32:14    May 20 2025 19:36:54

******** C96C48mx500_S2SW_cyc_gfs_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    May 20 2025 17:32:17    May 20 2025 17:52:07
202112201800        Done    May 20 2025 17:32:17    May 20 2025 19:57:40
202112210000        Done    May 20 2025 17:32:17    May 20 2025 20:56:48
202112211800        Done    May 20 2025 17:57:01    May 20 2025 20:56:48

******** C96C48_ufs_hybatmDA_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    May 20 2025 17:32:19    May 20 2025 17:52:12
202402240000        Done    May 20 2025 17:32:19    May 20 2025 20:26:49
202402240600        Done    May 20 2025 17:32:19    May 20 2025 20:12:02

******** C96mx100_S2S_3683 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
199405010000        Done    May 20 2025 17:32:22    May 20 2025 19:12:00
```

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/remove_compath_usage

DavidHuber-NOAA reviewed (approved):
Looks good!
